nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-error-handler.js

```diff
@@ -65,8 +65,20 @@
             // shut off the auto refresh
             nfCanvas.stopPolling();
 
-            // allow page refresh with ctrl-r
+            // disable page refresh with ctrl-r
             nfCanvas.disableRefreshHotKey();
+        },
```

Comment:
While not introduced here, I believe there may be a bug with the implementation of `handleAjaxError` here. Specifically, we only want to do these actions when the handling of the ajax error resulted in hiding the canvas. Wondering if we can simply return that fact from `handleAjaxError`? That function is used directly in jQuery Deferred APIs. I don't see any issue in their docs in doing this.

This issue will be more widespread now that more of the error handling will be funneling through these functions and no longer handled inline.

Reply:
Added an `if` condition before executing these actions to disable canvas, so that these are only executed when the message pane is shown.

I tested this works with modifying a NiFi REST API endpoint to throw NullPointerException. Js received 500 response and show the message pain as expected. Also, confirmed that `ctrl+r` hotkey is disabled.
![image](https://user-images.githubusercontent.com/1107620/56102890-e1254900-5f6a-11e9-9819-9f4e0d1cc51d.png)

This approach is not what you suggested above that uses jQuery Deferred API, but I hope I understand the requirements correctly and the updated code looks good.